book/crud.rst

```diff
@@ -0,0 +1,767 @@
+        {{ UI.saveAndCloseButton('Save'|trans) }}
+    {% endblock navButtons %}
+
+    {% block pageHeader %}
+        {% if form.vars.value.id %}
+            {% set breadcrumbs = {
+                'entity':      form.vars.value,
+                'indexPath':   path('inventory.vehicle_index'),
+                'indexLabel': 'Vehicles'|trans,
+                'entityTitle': form.vars.value.model
+            } %}
+            {{ parent() }}
+        {% else %}
+            {% set title = 'oro.ui.create_entity'|trans({'%entityName%': 'Vehicle'|trans}) %}
+            {% include 'OroUIBundle::page_title_block.html.twig' with { title: title } %}
+        {% endif %}
+    {% endblock pageHeader %}
+
+    {% block content_data %}
+        {% set id = 'vehicle-edit' %}
+
+        {% set dataBlocks = [{
+                'title': 'General'|trans,
+                'class': 'active',
+                'subblocks': [{
+                    'title': '',
+                    'data': [
+                        form_row(form.model),
+                        form_row(form.seats),
+                        form_row(form.boughtAt),
+                        form_row(form.leasedUntil),
+                    ]
+                }]
+            }]
+        %}
+        {% set data = {
+            'formErrors': form_errors(form)? form_errors(form) : null,
+            'dataBlocks': dataBlocks,
+        } %}
+        {{ parent() }}
+    {% endblock content_data %}
+
+The template extends the `update.html.twig`_ base template that is defined in the OroUIBundle. You
+just have to customize some important blocks to get desired output:
+
+``navButtons``
+
+    With this block, you can customize which buttons will be shown above the form allowing the user
+    to save the form and to trigger additional options. Beware that you set the ``formAction``
+    variable to a proper value depending on the controller the user accessed.
+
+``pageHeader``
+
+    This is the title block that is used to show a breadcrumb-style navigation and a custom title.
+
+``content_data``
+
+    This block defines the main contents of the update page which is read from the ``data``
+    variable by the parent template. If you want to divide your form into several sections, you can
+    define multiple ``dataBlocks`` here for this purpose.
+
+.. seealso::
+
+    Take a look at the templates that are shipped with the OroUIBundle to find out what blocks you
+    can also use to customize the layout of your forms.
+
+After the user submitted the form and the entered data is valid, the data will be persisted using
+the entity manager and a flash message is added which will be shown to the user on the request. The
+``oro_ui.router`` service is then used to redirect the user depending on the button they clicked to
+submit the form. They will be redirected to the first route that is passed to
+``redirectAfterSave()`` if they chose the *Save* button and if the user chose the *Save and close*
```

Comment:
Only 'Save' button (That acts as "Save and Close") is rendered using the provided code in `update.html.twig`. To have 2 buttons the code like this should be used instead:
```twig
    {% set html = UI.saveAndCloseButton() %}
    {% if resource_granted('inventory.vehicle_update') %}
        {% set html = html ~ UI.saveAndStayButton() %}
    {% endif %}
    {{ UI.dropdownSaveButton({'html': html}) }}
```